Exercises/NoModules/Chapter 07/Soln7_07.cpp

```diff
@@ -0,0 +1,43 @@
+// Exercise 7-7
+// Practice string concatenation, looping over strings, and stoi()-like functions
+#include <iostream>
+#include <string> 
+
+int main()
+{
+  std::string numbers;
+  
+  std::cout << "Enter a sequence of numbers terminated by #:\n";
+  
+  while (true)
+  {
+    std::string number;         // Read strings one by one
+    std::cin >> number;
+    if (number == "#") break;   // Stop indefinite loop once "#" is encountered
+    numbers += number + ' ';    // Concatenate all individual strings into one single string
+  }
```

Comment:
you could also use std::getline() instead of this whole while loop.
That would fix my bug that I had to enter # twice.
But then you need to adjust the below processing loop to allow not only a single space between numbers, but any amount of whitespace, including new lines.